### PR TITLE
gdk-pixbuf: modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
```
[162/165] Linking target gdk-pixbuf/gdk-pixbuf-pixdata
FAILED: gdk-pixbuf/gdk-pixbuf-pixdata 
clang @gdk-pixbuf/gdk-pixbuf-pixdata.rsp
gdk-pixbuf/libgdk_pixbuf-2.0.a(io-bmp.c.o): In function `DecodeHeader':
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-bmp.c:458: undefined reference to `__muloti4'
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-bmp.c:480: undefined reference to `__muloti4'
gdk-pixbuf/libgdk_pixbuf-2.0.a(io-gif.c.o): In function `gif_get_lzw':
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-gif.c:376: undefined reference to `__muloti4'
gdk-pixbuf/libgdk_pixbuf-2.0.a(io-gif-animation.c.o): In function `gdk_pixbuf_gif_anim_iter_get_pixbuf':
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-gif-animation.c:424: undefined reference to `__muloti4'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[163/165] Linking target gdk-pixbuf/gdk-pixbuf-query-loaders
[164/165] Linking target gdk-pixbuf/gdk-pixbuf-csource
FAILED: gdk-pixbuf/gdk-pixbuf-csource 
clang @gdk-pixbuf/gdk-pixbuf-csource.rsp
gdk-pixbuf/libgdk_pixbuf-2.0.a(io-bmp.c.o): In function `DecodeHeader':
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-bmp.c:458: undefined reference to `__muloti4'
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-bmp.c:480: undefined reference to `__muloti4'
gdk-pixbuf/libgdk_pixbuf-2.0.a(io-gif.c.o): In function `gif_get_lzw':
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-gif.c:376: undefined reference to `__muloti4'
gdk-pixbuf/libgdk_pixbuf-2.0.a(io-gif-animation.c.o): In function `gdk_pixbuf_gif_anim_iter_get_pixbuf':
/home/conan/w/prod/BuildSingleReference/.conan/data/gdk-pixbuf/2.42.4/_/_/build/f5b952277482ba74fd50e98a8432c4b7d50f5c48/build_subfolder/../source_subfolder/gdk-pixbuf/io-gif-animation.c:424: undefined reference to `__muloti4'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

https://bugs.llvm.org/show_bug.cgi?id=16404

**Review by uilianries — APPROVED** [review]

File: recipes/gdk-pixbuf/all/conanfile.py [hunk 1]
```diff
@@ -51,16 +56,6 @@ def configure(self):
...
-            raise ConanInvalidConfiguration("This package does not support Macos currently")
-
-    def build_requirements(self):
-        self.build_requires("meson/0.61.2")
-        self.build_requires("pkgconf/1.7.4")
-        if self.options.with_introspection:
-            self.build_requires("gobject-introspection/1.70.0")
 
     def requirements(self):
         self.requires("glib/2.70.4")
```
Comment (ericLemanissier) [inline recipes/gdk-pixbuf/all/conanfile.py:61]:
```suggestion
        self.requires("glib/2.72.0")
```

Comment (SpaceIm) [reply recipes/gdk-pixbuf/all/conanfile.py:61]: nop, see previous commits

Comment (ericLemanissier) [reply recipes/gdk-pixbuf/all/conanfile.py:61]: Please put a comment on this line, or someone(including me) will inevitably try to bump it later.

Comment (ericLemanissier) [reply recipes/gdk-pixbuf/all/conanfile.py:61]: Also, what is the interaction between glib and the LLVM issue posted above?

Comment (SSE4) [reply recipes/gdk-pixbuf/all/conanfile.py:61]: what is this LLVM issue?

Comment (ericLemanissier) [reply recipes/gdk-pixbuf/all/conanfile.py:61]: The one in https://github.com/conan-io/conan-center-index/pull/10154#issuecomment-1094224794

Comment (SSE4) [reply recipes/gdk-pixbuf/all/conanfile.py:61]: besides switching to `compiler-rt`, the workaround could be to define function `__muloti4`, e.g. like here:
https://opensource.apple.com/source/clang/clang-600.0.54/src/projects/compiler-rt/lib/muloti4.c.auto.html
the bug seems to be old enough

Comment (SpaceIm) [reply recipes/gdk-pixbuf/all/conanfile.py]: https://github.com/conan-io/conan-center-index/pull/10154/commits/d055044b85f68aa6a1b1076aa565b2e03d8a3c5a

**Comment by SpaceIm** [comment]
Well, let's keep previous version of glib unconditionally because it was not the goal this PR and I don't have the time to fix this.

**Comment by ericLemanissier** [comment]
ok, this is caught by the new hook
```
[HOOK - conan-center.py] post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library '.\bin\gio-2.0-0.dll' links to system library 'iphlpapi' but it is not in cpp_info.system_libs.
[HOOK - conan-center.py] post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library '.\bin\gio-2.0-0.dll' links to system library 'dnsapi' but it is not in cpp_info.system_libs.
[HOOK - conan-center.py] post_package(): WARN: [MISSING SYSTEM LIBS (KB-H043)] Library '.\bin\gio-2.0-0.dll' links to system library 'shlwapi' but it is not in cpp_info.system_libs.
```
I'll do a PR. We are in a bad situation with overriding of glib, with the change of default from shared to static starting with 2.71.1..

**Review by ericLemanissier — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
> ok, this is caught by the new hook

great, it provides some useful feedback, and doesn't appear to break anything. great job @madebr !

**Pull request merged**